### ISSUE TITLE
GraphQL: removeCouponFromCart mutation not resetting applied_rule_ids on quote item

### ISSUE THREAD

**Issue by Sekiphp** [magento/magento2#36887]
### Preconditions and environment

Magento version: 
```cli
$ ./bin/magento --version
Magento CLI 2.4.5
```
(successfully simulated also on Adobe Commerce 2.4.4)

### Steps to reproduce

1. Create cart price rule (coupon: specific coupon)
2. Create empty cart 
```graphql
mutation {
  createEmptyCart
}
```
4. Add product to cart (simple product)
5. Apply coupon on cart (successfully)
```graphql
mutation($cart_id: String!, $coupon_code: String!) {
    applyCouponToCart(input: {
        cart_id: $cart_id, 
        coupon_code: $coupon_code
    }) { 
        cart { 
            id 
            applied_coupons { 
                code 
            }
        }
    }
}
```

7. Remove coupon from cart
```graphql
mutation RemoveCouponFromCart($cart_id: String!) {
  removeCouponFromCart(input: {cart_id: $cart_id}) {
    cart {
      applied_coupons {
        code
      }
    }
  }
}
```

9. Check value of column `applied_rule_ids` in table table `quote_item` - there should be null or empty string, but sales_rule id is there.

### Expected result

`applied_rule_ids` in table `quote_item` should be empty after `removeCouponFromCart` mutation was called.

### Actual result

`applied_rule_ids` in table `quote_item` contains rule_id of previously used coupon (sales rule).
![image](https://user-images.githubusercontent.com/9967016/219984805-f18b1d48-73be-4505-aa6b-a4d75da48102.png)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
Hi @Sekiphp ,
Thank you for reporting and collaboration. Verified the issue on Magento 2.4-develop instance but the issue is not reproducible with above mentioned steps.
applied_rule_ids in quote_item DB table is returning null after removing coupon code using graphql
![image](https://user-images.githubusercontent.com/60198231/220242778-a8562f37-d5a6-4e18-a037-6adb8fcd8957.png)
![image](https://user-images.githubusercontent.com/60198231/220242822-295b9a39-6acd-4508-9da4-87ed32b4aaa2.png)
Kindly recheck the behavior on Magento 2.4-develop instance and provide missing steps if any if the issue is still reproducible.
Thank you.

**Comment by Sekiphp** [comment]
@engcom-November

Hello, I'm try to simulate again on clean environment and I successfully reproduced this bug again. 
Better reproduction steps:

1. Create cart price rule 
![image](https://user-images.githubusercontent.com/9967016/220887378-ee7e4ce5-5857-4280-b4aa-fb3a2f7dc592.png)
![image](https://user-images.githubusercontent.com/9967016/220887457-5bce4a72-ff83-4366-923c-67eb28392e58.png)
![image](https://user-images.githubusercontent.com/9967016/220887533-8341db94-833f-4b17-9914-ba8e7f60b6aa.png)

2. Create empty cart on default store
```graphql
mutation {
    createEmptyCart
}
```

3. Add simple product to cart as mentioned above. Output should be like:
```graphql
{
    "data": {
        "addProductsToCart": {
            "cart": {
                "items": [
                    {
                        "id": "1",
                        "uid": "MQ==",
                        "quantity": 2,
                        "product": {
                            "sku": "auo"
                        },
                        "prices": {
                            "row_total": {
                                "value": 2000,
                                "currency": "CHF"
                            }
                        }
                    }
                ]
            },
            "user_errors": []
        }
    }
}
```

4. Apply the coupon created in step 1

Input:
```graphql
mutation ApplyCouponToCart($cart_id: String!, $coupon_code: String!) {
    applyCouponToCart(input: {
        cart_id: $cart_id, 
        coupon_code: $coupon_code
    }) { 
        cart { 
            applied_coupons { 
                code
            }
            prices { 
                grand_total { 
                    value 
                    currency 
                }
                discounts {
                    amount {
                        value
                        currency
                    }
                    label
                }
            }
        }
    }
}
```

Result:
```graphql
{
    "data": {
        "applyCouponToCart": {
            "cart": {
                "applied_coupons": [
                    {
                        "code": "discount-100"
                    }
                ],
                "prices": {
                    "grand_total": {
                        "value": 1800,
                        "currency": "CHF"
                    },
                    "discounts": [
                        {
                            "amount": {
                                "value": 200,
                                "currency": "CHF"
                            },
                            "label": "Discount"
                        }
                    ]
                }
            }
        }
    }
}
```

5. Run SQL query
```sql
SELECT SQL_NO_CACHE item_id, quote_id, applied_rule_ids FROM quote_item;
```

Output should be like:
![image](https://user-images.githubusercontent.com/9967016/220888639-6196abe4-09a0-4d3a-a5e4-e0aaf67a6730.png)

6. Remove coupon from cart
```graphql
mutation RemoveCouponFromCart($cart_id: String!) {
  removeCouponFromCart(input: {cart_id: $cart_id}) {
    cart {
      applied_coupons {
        code
      }
      prices {
        grand_total {
          value
          currency
        }
        discounts {
          amount {
            currency
            value
          }
          label
        }
      }
    }
  }
}
```

Expected output:
```graphql
{
    "data": {
        "removeCouponFromCart": {
            "cart": {
                "applied_coupons": null,
                "prices": {
                    "grand_total": {
                        "value": 2000,
                        "currency": "CHF"
                    },
                    "discounts": null
                }
            }
        }
    }
}
```

5. Run SQL query
```sql
SELECT SQL_NO_CACHE item_id, quote_id, applied_rule_ids FROM quote_item;
```

Output should be like:
![image](https://user-images.githubusercontent.com/9967016/220889039-ca852df7-9c9b-4a42-8574-6dcc41d2d29e.png)

In quote is situation ok - coupons are reseted successfully there.

**Comment by Sekiphp** [comment]
@magento give me 2.4.5 instance

**Comment by abdel-aouby** [comment]
I confirm that, I was successfully reproducing it on a clean 2.4.5 version
the quote table is cleaned, however quote_item is not

**Comment by Sekiphp** [comment]
@engcom-November I created quote in Magento instance, but for me is there no way how to check the database. Can you?

**Comment by engcom-November** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-November** [comment]
Hi @Sekiphp ,
Thank you for the response. Verified the issue again on Magento 2.4-develop instance with the updated steps as per [comment](https://github.com/magento/magento2/issues/36887#issuecomment-1441566898) and the issue is reproducible.
applied_rule_ids from quote_item table is not becoming null after running removeCouponFromCart mutation. Hence updating description and confirming the issue.

**Comment by engcom-November** [comment]
Hi @Sekiphp ,
I re verified the issue again on latest Magento 2.4-develop instance but the issue is not reproducible with the [steps](https://github.com/magento/magento2/issues/36887#issuecomment-1441566898) provided by you.
Both "quote_item" and "quote" tables are returning null for "applied_rule_ids" after running RemoveCouponFromCart Graphql mutation.
![image](https://user-images.githubusercontent.com/60198231/222627471-bd312be7-6dc4-4053-897e-4bb389606965.png)
![image](https://user-images.githubusercontent.com/60198231/222627437-0b81da5b-8d71-4537-9975-b8aea73aa603.png)
![image](https://user-images.githubusercontent.com/60198231/222627542-39a72c29-4b0b-4e6b-8c92-f0b4d9cbbeb6.png)
Kindly recheck the issue on Magento 2.4-develop instance as it is having latest code base and upcoming 2.4.x release and provide missing steps if any if the issue is still reproducible.
Thank you.

**Comment by engcom-November** [comment]
Hi @Sekiphp ,
Sorry for the confusion. I was having issue with Magento instance when I tested 2 weeks ago and is not an updated one. Hence didn't confirm the issue. I retested the issue again on latest Magento 2.4-develop instance without any 3rd party / custom modules enabled and the issue is not reproducible. Its recommended to verify issues on clean Magento 2.4-develop instance as it is having latest code base and upcoming 2.4.x release. Kindly recheck the issue on Magento 2.4-develop instance and provide missing steps if any if the issue is still reproducible.
Thank you.
